variables.tf

```diff
@@ -1,11 +1,153 @@
+variable "additional_routes" {
+  default     = []
+  description = "A list of additional routes that should be attached to the Client VPN endpoint"
+
+  type = list(object({
+    destination_cidr_block = string
+    description            = string
+    target_vpc_subnet_id   = string
+  }))
+}
+
+variable "additional_security_groups" {
+  default     = []
+  description = "List of security groups to attach to the client vpn network associations"
+  type        = list(string)
+}
+
+variable "associated_subnets" {
+  type        = list(string)
+  description = "List of subnets to associate with the VPN endpoint"
+}
+
+variable "authorization_rules" {
+  type = list(object({
+    name                 = string
+    access_group_id      = string
+    authorize_all_groups = bool
+    description          = string
+    target_network_cidr  = string
+  }))
+  description = "List of objects describing the authorization rules for the client vpn"
+}
+
+variable "client_cidr_block" {
+  type        = string
+  description = "(optional) describe your variable"
+}
+
+variable "cloudwatch_log_retention_days" {
+  default     = 30
+  description = "How long to keep VPN logs. Possible values are: 1, 3, 5, 7, 14, 30, 60, 90, 120, 150, 180, 365, 400, 545, 731, 1827, 3653, and 0. If you select 0, the events in the log group are always retained and never expire."
+  type        = number
+
+  validation {
+    error_message = "Invalid value. Possible values are: 1, 3, 5, 7, 14, 30, 60, 90, 120, 150, 180, 365, 400, 545, 731, 1827, 3653, and 0. If you select 0, the events in the log group are always retained and never expire."
+
+    condition = (
+      contains(
+        [
+          1,
+          3,
+          5,
+          7,
+          14,
+          30,
+          60,
+          90,
+          120,
+          150,
+          180,
+          365,
+          400,
+          545,
+          731,
+          1827,
+          3653,
+          0
+        ],
+        var.cloudwatch_log_retention_days
+      )
+    )
+  }
+}
```

Comment:
This is cool but I'm more of the mind to let them try and fail

Reply:
You commented on the wrong lines...I assume you're talking about the validation, and I'm only ok with that when it fails at plan time. Apply time failures SUCK and I want to avoid them at all costs.